syntax = "proto3";

package meter;

service MeterService {
    rpc GetReadings (ReadingRequest) returns (ReadingResponse);
    rpc CreateReadings (MeterDataPayload) returns (ReadingCreateResponse);
}

message ReadingRequest {
    string start_date = 1;
    string end_date = 2;
    int32 meter_id = 3;
}

message MeterDataPayload {
    string meter_serial = 1;
    string timestamp = 2;
    int32 channel = 3;
    float value = 4;
    string source = 5;
    string reading_reason = 6;
}

message Reading {
    int32 value = 1;
    string created_at = 2;
    string source = 3;
}

message ReadingResponse {
    repeated Reading readings = 1;
}

message ReadingCreateResponse {
    int64 id = 1;
}